src/basic/chattr-util.c

```diff
@@ -172,3 +173,64 @@ int read_attr_at(int dir_fd, const char *path, unsigned *ret) {
 
         return read_attr_fd(fd, ret);
 }
+
+int get_proj_id(int fd, uint32_t *ret) {
+        assert(ret);
+
+        struct fsxattr attrs;
+
+        if (ioctl(fd, FS_IOC_FSGETXATTR, &attrs) < 0)
+                return -errno;
+
+        *ret = 0;
+
+        if (attrs.fsx_projid > 0)
```

Comment:
fsx_projid is unsigned so just assign it directly, if it's 0 then it's fine and the above initialization can be removed